sycl/include/sycl/sub_group.hpp

```diff
@@ -309,16 +309,16 @@ struct sub_group {
     multi_ptr<T, Space, IsDecorated> src =
         sycl::detail::GetUnqualMultiPtr(cv_src);
 #ifdef __SYCL_DEVICE_ONLY__
-#ifdef __NVPTX__
+#if defined(__NVPTX__) || defined(__AMDGCN__)
     return src.get()[get_local_id()[0]];
 #else
     return sycl::detail::sub_group::load(src);
-#endif // __NVPTX__
+#endif // __NVPTX__ || __AMDGCN__
 #else
     (void)src;
     throw runtime_error("Sub-groups are not supported on host device.",
                         PI_ERROR_INVALID_DEVICE);
-#endif
+#endif // __NVPTX__ || __AMDGCN__
```

Comment:
The comment here should say `// __SYCL_DEVICE_ONLY__`

Reply:
Fixed here https://github.com/intel/llvm/pull/10855